### DISCUSSION TITLE
What is the name of the chat reponse renderer that changes MD to HTML?

### DISCUSSION THREAD

**Discussion by JJRobertsSex30** [open-webui/open-webui#2670] [General]
Hey I noticed that this app changes the markdown formatted responses from the LLMs automatically into correctly formatted HTML on the fly as the responses stream back.

I know its not hard to find a component that does this with a full message in MD but have not found one that does it on the fly with a streamed response from the server.

What the name of this component and is it a separate and re-usable component or is it written for this app ?

**Comment by justinh-rahb** [comment]
`markedjs`: https://github.com/markedjs/marked

**Reply by JJRobertsSex30** [reply]
Does it support on-the-fly rendering automatically when you are streaming MD text to it from the server or do you need to do anything special to get it to convert ?

**Reply by justinh-rahb** [reply]
I'd suggest reading the documentation or asking your favourite LLM about how it is used, it'll be a much better answer than I could give.
